### PR TITLE
Adding Backup Capabilities for Clouds

### PR DESCRIPTION
**Pull request by HoustonPutman**
*Issue number of the reported bug or feature request: #2 *

**Describe your changes**
I have implemented backups that can be persisted to a Kubernetes volume or to an s3 compatible endpoint.

**Testing performed**
Testing is done manually for now. Will add better tests.

**Additional context**
Add any other context about your contribution here.

### CODE REVIEW

File: example/ext_ops.yaml [hunk 1]
```diff
@@ -112,7 +112,7 @@ spec:
             value: "zk-operator"
 
 ---
-apiVersion: v1beta1
+apiVersion: v1
```
Comment (sepulworld) [inline example/ext_ops.yaml]: 👍 I ran into this being a problem when setting up locally. going to v1 on k8s 1.14.3 worked for me too.

Comment (HoustonPutman) [reply example/ext_ops.yaml]: Yeah thought I would just include those fixes in my big PR, as those files are auto-generated, so I needed to fix the source of those values.

Thanks for pointing it out!

**Comment by sepulworld** [comment]
@HoustonPutman question on how you used kubebuilder to create scaffolding for additional CRD.

I'm looking to use it for the SolrCloudCollections CRD

```
cd solr-operator
kubebuilder create api --group 'solr.bloomberg.com' --version v1beta1 --kind SolrCloudCollection
Create Resource [y/n]
y
Create Controller [y/n]
y
Writing scaffold for you to edit...
2019/08/05 13:03:59
```

But this doesn't generate any scaffolding. Curious what you are doing to get a new CRD started

**Comment by sepulworld** [comment]
```
$ kubebuilder version
Version: version.Version{KubeBuilderVersion:"2.0.0-beta.0", KubernetesVendor:"1.14.1", GitCommit:"b12371ccabd1bc8de7ec7ec1a1ca7bfabe567701", BuildDate:"2019-07-19T19:02:15Z", GoOs:"unknown", GoArch:"unknown"}
```

**Comment by sepulworld** [comment]
Nevermind... I just saw `hack/install_dependencies.sh` and my kubebuilder version might be too new.

**Comment by HoustonPutman** [comment]
the group should be just 'solr' not 'solr.bloomberg.com' .

Actually just answered this for myself when creating the prometheus exporter CRD.

**Comment by swarupdonepudi** [comment]
@HoustonPutman what is the prometheus exporter CRD that you were referring to in your previous comment? I just started exploring how to get the metrics for solr cloud deployed using this operator on to datadog.

**Pull request merged**